ENIGMAsystem/SHELL/Widget_Systems/widgets_mandatory.h

```diff
@@ -22,22 +22,46 @@
 #include "libEGMstd.h"
 
 #include <string>
-using std::string;
 
-namespace enigma
-{
+#define DEBUG_MESSAGE(msg, severity) show_error((std::string)msg + " | " + __FILE__ + ":" + std::to_string(__LINE__), severity)
+
+namespace enigma {
+  enum MESSAGE_TYPE : int {
+    M_INFO = 0,
+    M_WARNING = 1,
+    M_ERROR = 2,
+    M_USER_ERROR = 3,
+    M_FATAL_ERROR = 4,
+    M_FATAL_USER_ERROR = 5
+  };
+  
+  inline std::string error_type(MESSAGE_TYPE t) {
+    switch(t) {
+      case M_INFO: return "INFO";
+      case M_WARNING: return "WARNING";
+      case M_ERROR: return "ERROR";
+      case M_USER_ERROR: return "USER_ERROR";
```

Comment:
It's worth pointing out that in places where you've used this, it's invariably been some form of argument error. Which then begs another question: should we include different classes of exception in this enum? No strong feelings, personally.